command_line/dials_import.py

```diff
@@ -739,215 +742,235 @@ def convert_to_grid_scan(self, imageset_list, params):
-        self.write_experiments(experiments, params)
+    # Only allow a single sequence
+    if params.input.allow_multiple_sequences is False:
+        assert_single_sequence(experiments, params)
 
-    def write_experiments(self, experiments, params):
-        """
-        Output the experiments to file.
-        """
-        if params.output.experiments:
-            logger.info("-" * 80)
-            logger.info("Writing experiments to %s", params.output.experiments)
-            experiments.as_file(
-                params.output.experiments, compact=params.output.compact
-            )
+    # Write the experiments to file
+    write_experiments(experiments, params)
+    return experiments
 
-    def assert_single_sequence(self, experiments, params):
-        """
-        Print an error message if more than 1 sequence
-        """
-        sequences = [
-            e.imageset for e in experiments if isinstance(e.imageset, ImageSequence)
-        ]
-
-        if len(sequences) > 1:
-
-            # Print some info about multiple sequences
-            self.diagnose_multiple_sequences(sequences, params)
-
-            # Raise exception
-            raise Sorry(
-                """
-        More than 1 sequence was found. Two things may be happening here:
-
-        1. There really is more than 1 sequence. If you expected this to be the
-           case, set the parameter allow_multiple_sequences=True. If you don't
-           expect this, then check the input to dials.import.
-
-        2. There may be something wrong with your image headers (for example,
-           the rotation ranges of each image may not match up). You should
-           investigate what went wrong, but you can force dials.import to treat
-           your images as a single sequence by using the template=image_####.cbf
-           parameter (see help).
-      """
-            )
 
-    def diagnose_multiple_sequences(self, sequences, params):
-        """
-        Print a diff between sequences.
-        """
-        logger.info("")
-        for i in range(1, len(sequences)):
-            logger.info("=" * 80)
-            logger.info("Diff between sequence %d and %d", i - 1, i)
-            logger.info("")
-            self.print_sequence_diff(sequences[i - 1], sequences[i], params)
-        logger.info("=" * 80)
-        logger.info("")
+@show_mail_handle_errors()
+def run(args=None, *, phil=phil_scope):
+    do_import(args, phil=phil, configure_logging=True)
 
-    @staticmethod
-    def print_sequence_diff(sequence1, sequence2, params):
-        """
-        Print a diff between sequences.
-        """
-        logger.info(
-            compare.sequence_diff(
-                sequence1, sequence2, tolerance=params.input.tolerance
-            )
+
+class ImageImporter:
+    def __init__(self, phil=phil_scope) -> None:
+        warnings.warn(
+            "ImageImporter class is deprecated. Please use dials.command_line.dials_import.run instead",
```

Comment:
"Please use dials.command_line.dials_import.run"
Why not dials.command_line.dials_import.do_import? 

It looks like you fixed the only usage of `ImageImporter`, the one in `test_generate_mask`, so this is mostly a useless comment.

Reply:
Reasoning:
1. `.do_import` doesn't do the same thing as `ImageImporter` did, but `run` does - it assumes you have done no pre-parsing, so it's the smallest change that would give the same results.
2. It's the only usage Inside DIALS - I cannot discount the possibility that someone else is using `ImageImporter`, as a public API, to wrap image importing. This might be overly cautious, but it costs nothing to have a transition period. This does, however, point out that I didn't make a "Remove after X.Y" comment alongside this deprecation, so at least that needs to change.

Reply:
Scratch #1, I was remembering the reasoning from an earlier stage of the branch. `do_import` would indeed have the same behaviour as importing and using, because it wouldn't configure logging (whereas `.run` would). So that's a good catch.